pkg/cmd/sniff.go

```diff
@@ -166,7 +173,25 @@ func (o *Ksniff) Complete(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	o.restConfig, err = o.configFlags.ToRESTConfig()
+	var currentContext *api.Context
+	var exists bool
+
+	if o.settings.UserSpecifiedKubeContext != "" {
+		currentContext, exists = o.rawConfig.Contexts[o.settings.UserSpecifiedKubeContext]
+	} else {
+		currentContext, exists = o.rawConfig.Contexts[o.rawConfig.CurrentContext]
+	}
+
+	if !exists {
+		return errors.New("context doesn't exist")
+	}
+
+	o.restConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: o.configFlags.ToRawKubeConfigLoader().ConfigAccess().GetDefaultFilename()},
```

Comment:
Can you please explain the purpose of this line? why we should specify the path of the kubectl config file? isn't the API perform this lookup on his own?

Reply:
Actually I thought the same thing but it didn't work without specifying that parameter.